src/readable_compacts_transform_helper.js

```diff
@@ -0,0 +1,274 @@
+    let qualifiedMsgName = getQualifiedMsgName(msgName, pkgName);
+    let field = getField({
+        'name': matched[0]
+    }, qualifiedMsgName, messageFieldsMapping);
+    let result = [comToLong(field['id'])];
+    if (remainString[0] == '[') {
+        let indexOfClose = findEndBracket(remainString);
+        let str = remainString.substring(1, indexOfClose);
+        remainString = remainString.substring(indexOfClose + 1);
+        if (field['rule'] === 'repeated') {
+            if (str.match(/^[0-9]+$/)) {
+                result = result.concat([comToLong(parseInt(str))]);
+            } else {
+                throw new Error(str + ' is not a valid index for repeated item.');
+            }
+        }
+        else if (field['rule'] === 'map'){
+            switch (field['keytype']){
+            case 'string':
+                result = result.concat(strToComponents(str));
+                break;
+            case 'bytes':
+                result = result.concat(hexStrToComponents(str));
+                break;
+            case 'int64':
+            case 'int32':
+            case 'sint64':
+            case 'sint32':
+                result.push(Long.fromString(str,false));
+                break;
+            case 'uint64':
+            case 'uint32':
+            case 'fixed64':
+            case 'fixed32':
+                result.push(Long.fromString(str,true));
+                break;
+            default:
+                throw new Error(`Invalid key type: ${field['keytype']}`);
+            }
+        }else {
+            throw new Error(`Invalid field type: ${field['rule']} enclosed by [].`);
+        }
+    } 
+    if (remainString[0] == '.') {
+        remainString = remainString.substring(1);
+    }
+    result =  result.concat(doReadableCompactsMapping(messageFieldsMapping, field['type'], pkgName, remainString));
+    return result;
+};
+
+const buildMessageNameToFieldsMapping = (jsonMeta, results) => {
+    let scope = jsonMeta['package'];
+    if ((scope == null) || (scope == undefined)) {
+        buildNestedMessageNameToFieldsMapping('', jsonMeta, results);
+    } else {
+        buildNestedMessageNameToFieldsMapping(scope, jsonMeta, results);
+    }
+};
+
+const buildNestedMessageNameToFieldsMapping = (scope, jsonMeta, results) => {
+    let outer_scope = scope;
+    let name = jsonMeta['name'];
+    if (name != undefined) {
+        if (scope == '') {
+            outer_scope = name;
+        } else {
+            outer_scope = outer_scope + '.' + name;
+        }
+    }
+    if (jsonMeta['isNamespace'] == true) {
+        let msgs = jsonMeta['messages'];
+        for (let ii in msgs) {
+            buildNestedMessageNameToFieldsMapping(outer_scope, msgs[ii], results);
+        }
+    } else {
+        let msgs = jsonMeta;
+        for (let ii in msgs) {
+            results.set(outer_scope, msgs[ii]);
+        }
+    }
+};
+
+export class TransformHelper {
```

Comment:
Could you add [jsdoc](http://usejsdoc.org/about-getting-started.html) for the public method and add some comments here?

Reply:
will when most of the things is finalized